ftlib/__init__.py

```diff
@@ -26,29 +36,50 @@ def __init__(self):
         self.consensus = None
         self.framework = None
 
+    def _initialized_passive_check(self):
+        return self._initialized
+
+    def _initialized_active_check(self):
+        self.consensus.confirm()
+        return self._initialized
+
+    def _skip_allreduce_passive_check(self):
+        return self._skip_allreduce
+
+    def _skip_allreduce_active_check(self):
+        self.consensus.confirm()
+        return self._skip_allreduce
+
     def lock(self):
-        logging.debug(
-            "trying to lock with lock count: {}".format(self.lock_count)
-        )
+        logging.debug('trying to lock with lock count: {}'.format(
+            self.lock_count))
         self._lock.acquire()
         self.lock_count = self.lock_count + 1
         logging.debug("after look, lock count: {}".format(self.lock_count))
 
     def unlock(self):
-        logging.debug(
-            "trying to unlock with lock count: {}".format(self.lock_count)
-        )
+        logging.debug('trying to unlock with lock count: {}'.format(
+            self.lock_count))
         self._lock.release()
         self.lock_count = self.lock_count - 1
         logging.debug("after unlock, lock count: {}".format(self.lock_count))
 
-    def initialized(self):
-        return self._initialized
-
-    def init(self, consensus, framework):
-        assert framework in ["dummy_NCCL", "pytorch"]
-        self.consensus = SharedStorage(self)
-        if framework == "dummy_NCCL":
+    def init(self,
+             consensus,
+             framework,
+             consensus_init_kwargs=None,
+             framework_init_kwargs=None):
+        assert framework in framework_list
+        assert consensus in consensus_list
+
+        if consensus == 'shared_storage':
+            from .consensus.shared_storage import SharedStorage
+            self.consensus = SharedStorage(self)
+        if consensus == 'gossip':
```

Comment:
Can be `elif` here

Reply:
And https://github.com/caicloud/ftlib/pull/16/files#diff-23f25df454448e561799fb5c3dbc9f66R86